pages/index.js

```diff
@@ -3,28 +3,72 @@ import AddTask from '../components/AddTask'
 import { useEffect, useState } from 'react'
 import axios from '../utils/axios'
 import { useAuth } from '../context/auth'
+import { useRouter } from 'next/router'
+import Router from 'next/router'
+
 
 export default function Home() {
   const { token } = useAuth()
   const [tasks, setTasks] = useState([])
+  const router = useRouter()
+  const [isLoggedIn,setLoggedIn] = useState(false)
+
 
   function getTasks() {
-    /***
-     * @todo Fetch the tasks created by the user.
-     * @todo Set the tasks state and display them in the using TodoListItem component
-     * The user token can be accessed from the context using useAuth() from /context/auth.js
-     */
+    axios({
+      headers: {
+          Authorization: 'Token ' + token,
+      },
+      url:  'todo/',
+      method: 'get',
+    }).then(function({data, status}) { 
+      setTasks(data);
+    })
+
+  }
+
+  function deleteTask(id){
+    setTasks(prevTasks => {
+      return prevTasks.filter((task) => {
+        return task.id !== id;
+      });
+    });
+
   }
 
-  return (
+
+
+  useEffect(() => {
+    
+    if(token!="null"&&token!=undefined){
+      getTasks()
+      setLoggedIn(true)
+    }
+
+    const {pathname} = Router
+        if(pathname == '/' && (token=="null" || token==undefined)){
+            router.push('/login')
+        }
```

Comment:
You should have kept the code for authentication separate like in the middleware directory